flake.nix

```diff
@@ -23,31 +23,49 @@
           nixos-rebuild
           awscli2
           terraform
+          packer
           tailscale
           infracost
           deploy-rs.packages.${system}.default
+          jq
+          curl
         ];
       };
     });
 
     nixosConfigurations.factorio = nixpkgs.lib.nixosSystem {
       system = "x86_64-linux";
-      modules = [ ./configuration.nix ];
+      modules = [ ./configuration.nix ./factorio.nix ];
       specialArgs = { inherit impermanence; };
     };
 
     deploy.nodes.factorio = {
-      # By default, connect to host factorio with ssh user hacdc and sudo as root
-      # Note that for terraform deployments the sshUser and hostname are overridden.
+      # By default, connect to host factorio with ssh user hacdc and sudo as
+      # root Note that for terraform deployments the sshUser and hostname are
+      # overridden.
       hostname = "factorio";
-      sshUser = "hacdc";
+      sshUser = "root";
```

Comment:
Are we planning on disabling root login entirely?

Reply:
So this line was a workaround for signed paths. A `copy-closure` only works if you do a sign on the resulting build and the remote trusts the signature. I just hacked around it by copying the closure as root and dodging the issue.

As for root login: Maybe? There are two problems:
- For the amazon init, the ec2 metadata key is loaded into the root account [here](https://github.com/NixOS/nixpkgs/blob/f73008d3b9536cc3b235b2cc5bff8738758935d0/nixos/modules/virtualisation/ec2-data.nix#L34)
- Tailscale seems to have root login capabilities (I reconfigured the deploy to use tailscale) so... not sure what we do about that.

Reply:
All told, with passwordless sudo not sure it matters anyways since if you have access to a wheel account you have root anyways

Reply:
> Tailscale seems to have root login capabilities (I reconfigured the deploy to use tailscale) so... not sure what we do about that.

We can control who has access to what accounts via Tailscale ACL rules (https://tailscale.com/kb/1018/acls/), such as limiting users to accessing the `autogroup:nonroot` user

Tailscale ACLs are definitely a next step I'll be exploring in the future, I believe we can specify ACL rules in terraform but I haven't tested it.

Reply:
> So this line was a workaround for signed paths. A copy-closure only works if you do a sign on the resulting build and the remote trusts the signature. I just hacked around it by copying the closure as root and dodging the issue.

Could this be because `hacdc` is not part of the trusted users? If so this could fix the issue:
```nix
nix.settings.trusted-users = [ "@wheel" ];
```
(I have this line in my server deployments as well)

Reply:
D'oh. Yeah that would probably fix it